rj_gameplay/stp/skill/__init__.py

```diff
@@ -0,0 +1,77 @@
+from abc import ABC, abstractmethod
+from typing import Type, TypeVar, MutableMapping, Dict
+
+import stp.role as role
+
+
+class ISkill(ABC):
+    @abstractmethod
+    def define(self):
+        ...
+
+    @abstractmethod
+    def create_request(self) -> role.RoleRequest:
+        ...
+
+
+SkillT = TypeVar("SkillT", bound=ISkill)
+
+
+class Registry(MutableMapping):
```

Comment:
`SkillRegistry`?

Reply:
I was thinking about this, and because we're only doing `import .. as ...` calls, it's basically the difference between doing `skill.Registry` vs `skill.SkillRegistry`. I thought that the former was nice, but idk.

Reply:
True...I am 90% sure someone will `from skill import Registry` at some point, though, and I'd rather not constantly police for that.

Reply:
That breaks hot reload though... I think maybe I'll make a custom pylint checker that errors on any `from stp.* import .*` if it's not a test.

Reply:
Oh shit, you right. Yeah that sounds like a better path.